### ISSUE TITLE
Running bjoern loops in parallel

### ISSUE THREAD

**Issue by CMCDragonkai** [jonashaag/bjoern#141]
Although this is not in the docs anywhere, in one of your tests, you have `os.fork()` and you're running `bjoern.run()` multiple times.

What does this actually do?

Is the `bjoern` event loop single threaded (but it is not event driven like Node.js)? So does that mean normally it can only handle 1 request at a time, because the WSGI app will not yield back to the python scheduler? Therefore to get the best performance we need to run multiple `bjoern.run()` so that requests can be handled in parallel in multiple Python processes.

I'm hoping to get more documentation on the exact behaviour of bjoern in the presence of Python multithreading and its GIL lock, and multiprocessing as in the case of using `os.fork()`.

**Comment by jonashaag** [comment]
On Linux the kernel will round-robin the subprocesses, see e.g. https://stackoverflow.com/questions/9032445/what-happens-when-a-tcp-server-binds-and-forks-before-doing-an-accept-which-pr

**Comment by CMCDragonkai** [comment]
I decided to try this style instead of forking:

```
    with Pool(processes=worker_count) as pool:
        pool.apply_async(bjoern.run)
        pool.close()
        pool.join()
```

But while it worked for simple apps, it didn't work for my keras/tensorflow code. Seems like keras/tensorflow might have some global locking or something else.

**Comment by CMCDragonkai** [comment]
@jonashaag Can you clarify this question:

> Is the bjoern event loop single threaded (but it is not event driven like Node.js)? So does that mean normally it can only handle 1 request at a time, because the WSGI app will not yield back to the python scheduler? Therefore to get the best performance we need to run multiple bjoern.run() so that requests can be handled in parallel in multiple Python processes.

**Comment by jonashaag** [comment]
Bjoern is single threaded and evented, so if you Python application yields on a regular basis you may be okay with running only a single process. Bjoern does not mess around with the Python primitives (socket API etc) to enforce yields like other WSGI servers do. It also does not use Python's native asyncio Interface because it predates asyncio. If your application seldom yields, you probably should run multiple instances for parallel execution.
